packages/office-ui-fabric-react/src/utilities/dateMath/DateMath.ts

```diff
@@ -198,4 +198,46 @@ function getDatePartHashValue(date: Date) {
   /* tslint:disable:no-bitwise */
   return date.getDate() + (date.getMonth() << 5) + (date.getFullYear() << 9);
   /* tslint:enable:no-bitwise */
-}
\ No newline at end of file
+}
+
+/**
+ * Returns week number for a week in a year.
+ */
+export function getWeekNumbers(weeks: any[], firstDayOfWeek: DayOfWeek, navigatedDate: Date) {
+  let monthRange = getDateRangeArray(navigatedDate, DateRangeType.Month, firstDayOfWeek);
+
+  let getDayofYear = getDayOfYear(monthRange[0]);
+  let firstWeekNumber = Math.ceil(getDayofYear / DAYS_IN_WEEK);
+
+  let weeksArray = [];
+
+  for (let i = 0; i < weeks.length; i++) {
+    weeksArray.push(firstWeekNumber + i);
+  }
+
+  return weeksArray;
+}
+
+/**
+ * Returns the day number for a date in a year
+ * The number of days since January 1st in the particular year.
+ */
+export function getDayOfYear(date: Date) {
+  let month = date.getMonth();
+  let daysUntilDate = 0;
+
+  for (let i = 0; i < month; i++) {
+    daysUntilDate += daysInMonth(i, date.getFullYear());
+  }
+
+  daysUntilDate += date.getDate();
+
+  return daysUntilDate;
+}
+
+/**
+ * Returns the number of days in the month
+ */
+function daysInMonth(month: number, year: number) {
+  return new Date(year, month + 1, 0).getDate();
```

Comment:
Is date == 0 a supported value or are we relying on a side effect here? JSMVVM does this differently and has code to handle leap years.

Reply:
This already handles leap years.  I tried 2/2016 and got 29 days, 2/2017 (non-leap year) has 28 days.